File: python/novax_sdk/pay/h5/h5_ok_time_request.py
from dataclasses import dataclass
from typing import Any

from novax_sdk.request.api_request import ApiRequest, HttpMethod


@dataclass
class H5OkTimeRequest(ApiRequest):
    """POST /pay/public/h5/ok-time"""

    protocol: str | None = None
    currency: str | None = None
    smart_contract_address: str | None = None
    company_user_id: str | None = None
    token: str | None = None

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.POST

    @property
    def path(self) -> str:
        return "/pay/public/h5/ok-time"

    def body(self) -> dict[str, Any]:
        return {
            "protocol": self.protocol,
            "currency": self.currency,
            "smartContractAddress": self.smart_contract_address,
            "companyUserId": self.company_user_id,
            "token": self.token,
        }

    @property
    def response_type(self) -> type:
        return bool

File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/h5/H5OkTimeRequest.java
package com.novax.sdk.pay.h5;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * POST /pay/public/h5/ok-time — confirm the time at which payment was made.
 */
@Getter
@Builder
public class H5OkTimeRequest extends AbstractApiRequest<Boolean> {

    private final String protocol;
    private final String currency;
    private final String smartContractAddress;
    private final String companyUserId;
    private final String token;

    @Override
    public HttpMethod method() {
        return HttpMethod.POST;
    }

    @Override
    public String path() {
        return "/pay/public/h5/ok-time";
    }

    @Override
    public Object body() {
        Map<String, Object> b = new LinkedHashMap<>();
        b.put("protocol", protocol);
        b.put("currency", currency);
        b.put("smartContractAddress", smartContractAddress);
        b.put("companyUserId", companyUserId);
        b.put("token", token);
        return b;
    }

    @Override
    public TypeRef<Boolean> responseType() {
        return TypeRef.of(Boolean.class);
    }
}
